client/client.go

```diff
@@ -69,7 +70,7 @@ func NewFileCachedRepository(baseDir string, gun data.GUN, baseURL string, rt ht
 	if err != nil {
 		return nil, err
 	}
-
+	pkcs11.Setup()
```

Comment:
You are absolutely right. A non_pkcs11 package should not contain any relation to a pkcs11 package it. I added an init() function in the pkcs11 package.  
The init() just calls Setup(), because Docker for example uses another entrypoint into notary while building/pushing. It uses client/repo(_pkcs11).go as entry. If not calling Setup() there, the whole pkcs11 package would not be imported just pkcs11/common. 
So init of pkcs11 would  not be called. 
Therefore in repo_pcks11.go the call for Setup() remains.